src/lazyload-image.directive.ts

```diff
@@ -1,44 +1,61 @@
 import 'rxjs/add/operator/let';
-import { Directive, ElementRef, Input, NgZone } from '@angular/core';
-import { getScrollListener } from './scroll-listener';
-import { lazyLoadImage } from './lazyload-image';
+import {Directive, ElementRef, Input, NgZone, SimpleChanges, OnChanges, OnDestroy} from '@angular/core';
+import {getScrollListener} from './scroll-listener';
+import {lazyLoadImage} from './lazyload-image';
+import {WindowService} from "./window.service";
+
+declare const global: any; // To make AoT compiler (ngc) happy
 
 @Directive({
-    selector: '[lazyLoad]'
+  selector: '[lazyLoad]'
 })
-export class LazyLoadImageDirective {
-    @Input('lazyLoad') lazyImage;   // The image to be lazy loaded
-    @Input() errorImage: string;    // The image to be displayed if lazyImage load fails
-    @Input() scrollTarget = window; // Chnage the node we should listen for scroll events on, default is window
-    @Input() scrollObservable;      // Pass your own scroll emitter
-    @Input() offset: number;        // The number of px a image should be loaded before it is in view port
-    elementRef: ElementRef;
-    ngZone: NgZone;
-    scrollSubscription;
-
-    constructor(el: ElementRef, ngZone: NgZone) {
-        this.elementRef = el;
-        this.ngZone = ngZone;
-    }
+export class LazyLoadImageDirective implements OnChanges, OnDestroy {
+  @Input('lazyLoad') lazyImage;   // The image to be lazy loaded
+  @Input() errorImage: string;    // The image to be displayed if lazyImage load fails
+  @Input() scrollTarget; // Change the node we should listen for scroll events on, default is window
+  @Input() scrollObservable;      // Pass your own scroll emitter
+  @Input() offset: number;        // The number of px a image should be loaded before it is in view port
+  elementRef: ElementRef;
+  ngZone: NgZone;
+  scrollSubscription;
+  window;
+
+  constructor(el: ElementRef, ngZone: NgZone, window: WindowService) {
+    this.elementRef = el;
+    this.ngZone = ngZone;
+    // Get window object, fallback for Universal
+    this.window = window.nativeWindow || (<any>global);
+    this.scrollTarget = this.window;
```

Comment:
This will overwrite the user input for the `scrollTarget`. I guess we should do something like:
```js
this.window = window.nativeWindow || global;
this.scrollTarget = this.scrollTarget || this.window;
```